### PR TITLE
Add changelog for alertmanager sharding and add experimental warning log.

### PR DESCRIPTION
**Pull request by stevesg**
**What this PR does**:
- Add changelog for alertmanager sharding now the implementation is complete
- Add experimental warning log for the feature

**Checklist**
- [x] ~~Tests updated~~
- [x] ~~Documentation added~~
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pstibrany — APPROVED** [review]
Great job working on this feature! 🎉

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,11 @@
...
+  * Ensure that a remote storage backend is configured for Alertmanager to store state using `-alertmanager-storage.backend`, and flags related to the backend. Note that the `local` and `configdb` storage backends are not supported.
+  * Ensure that a ring store is configured using `-alertmanager.sharding-ring.store`, and set the flags relevant to the chosen store type.
+  * Enable the feature using `-alertmanager.sharding-enabled`.
+  * Note the prior addition of a new configuration option `-alertmanager.persist-interval`. This sets the interval between persisting the current alertmanager state (notification log and silences) to object storage. See the (configuration file reference)[(here for alertmanager)[https://cortexmetrics.io/docs/configuration/configuration-file/#alertmanager_config]] for more information.
```
Comment (pracucci) [inline CHANGELOG.md:18]: Did you mean this?

```suggestion
  * Note the prior addition of a new configuration option `-alertmanager.persist-interval`. This sets the interval between persisting the current alertmanager state (notification log and silences) to object storage. See the [configuration file reference](https://cortexmetrics.io/docs/configuration/configuration-file/#alertmanager_config) for more information.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,11 @@
...
 * [FEATURE] Alertmanager: Added `-alertmanager.max-config-size-bytes` limit to control size of configuration files that Cortex users can upload to Alertmanager via API. This limit is configurable per-tenant. #4201
 * [FEATURE] Added flag `-debug.block-profile-rate` to enable goroutine blocking events profiling. #4217
+* [FEATURE] Alertmanager: The experimental sharding feature is now considered complete. Detailed information about the configuration options can be found (here for alertmanager)[https://cortexmetrics.io/docs/configuration/configuration-file/#alertmanager_config] and (here for the alertmanager storage)[https://cortexmetrics.io/docs/configuration/configuration-file/#alertmanager_storage_config]. To use the feature:  #4222
```
Comment (pracucci) [inline CHANGELOG.md:14]: Did you mean this?

```suggestion
* [FEATURE] Alertmanager: The experimental sharding feature is now considered complete. Detailed information about the configuration options can be found [here for alertmanager](https://cortexmetrics.io/docs/configuration/configuration-file/#alertmanager_config) and [here for the alertmanager storage](https://cortexmetrics.io/docs/configuration/configuration-file/#alertmanager_storage_config). To use the feature:  #4222
```

Comment (pracucci) [inline CHANGELOG.md:14]: > To use the feature:  #4222

Could you add the PR numbers of all other PRs that build up the feature, please?

File: CHANGELOG.md (hunk 1 above)
Comment (stevesg) [reply CHANGELOG.md:18]: Doh :sleeping:

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,12 @@
...
 * [FEATURE] Alertmanager: Added `-alertmanager.max-config-size-bytes` limit to control size of configuration files that Cortex users can upload to Alertmanager via API. This limit is configurable per-tenant. #4201
 * [FEATURE] Added flag `-debug.block-profile-rate` to enable goroutine blocking events profiling. #4217
+* [FEATURE] Alertmanager: The experimental sharding feature is now considered complete. Detailed information about the configuration options can be found [here for alertmanager](https://cortexmetrics.io/docs/configuration/configuration-file/#alertmanager_config) and [here for the alertmanager storage](https://cortexmetrics.io/docs/configuration/configuration-file/#alertmanager_storage_config). To use the feature: #3925 #4020 #4021 #4031 #4084 #4110 #4126 #4127 #4182 #4141 #4146 #4162 #4222
```
[the remark this answers is not in the document]
Comment (stevesg) [reply CHANGELOG.md:14]: Not AFAIK (?)

[the remark this answers is not in the document]
Comment (stevesg) [reply CHANGELOG.md]: Ha, right! I meant #4161, though I'm not sure how I ended up there....

**Pull request merged**